candle-transformers/src/models/gemma3.rs

```diff
@@ -370,6 +378,43 @@ impl DecoderLayer {
     }
 }
 
+fn prepare_decoder_attention_mask(
+    b_size: usize,
+    tgt_len: usize,
+    seqlen_offset: usize,
+    is_sliding: bool,
+    sliding_window: usize,
```

Comment:
I think I slightly prefer the old implementation where `sliding_window` is an `Option<usize>`. Feels a bit better than having a boolean to control whether `sliding_window` gets used or not and it can be called twice below, once with `None`, and once with `Some(...)`.